pkg/storage/ingest/fetcher.go

```diff
@@ -388,6 +390,41 @@ func (r *concurrentFetchers) PollFetches(ctx context.Context) (kgo.Fetches, cont
 	}
 }
 
+func instrumentGaps(gaps []offsetRange, records prometheus.Counter, logger log.Logger) {
+	for _, gap := range gaps {
+		level.Error(logger).Log(
+			"msg", "there is a gap in consumed offsets; it is likely that there was data loss; see runbook for MimirIngesterMissedRecordsFromKafka",
+			"gap_start_inclusive", gap.start,
+			"gap_end_inclusive", gap.end,
```

Comment:
The `offsetRange` doc says that `end` is exclusive:

```suggestion
			"records_offset_gap_end_exclusive", gap.end,
```